src/MICore/Transports/PipeTransport.cs

```diff
@@ -361,5 +300,66 @@ private void OnProcessExit(object sender, EventArgs e)
                 }
             }
         }
+
+        private int WrappedExecuteSyncCommand(string commandDescription, string commandText, int timeout)
+        {
+            int exitCode = -1;
+            string output = null;
+            string error = null;
+
+            string fullCommand = string.Format(CultureInfo.InvariantCulture, "{0} {1} {2}", _pipePath, _cmdArgs, commandText);
+
+            try
+            {
+                exitCode = ExecuteSyncCommand(commandDescription, commandText, timeout, out output, out error);
+
+                if (exitCode != 0)
+                {
+                    this.Callback.OnStdErrorLine(string.Format(CultureInfo.InvariantCulture, MICoreResources.Warn_ProcessExit, fullCommand, exitCode));
+                }
+            }
+            catch (Exception e)
+            {
+                this.Callback.OnStdErrorLine(string.Format(CultureInfo.InvariantCulture, MICoreResources.Warn_ProcessException, fullCommand, e.Message));
+            }
+
+            return exitCode;
+        }
+
+        public override int ExecuteSyncCommand(string commandDescription, string commandText, int timeout, out string output, out string error)
+        {
+            output = null;
+            error = null;
+            int exitCode = -1;
+
+            Process proc = new Process();
+            try
+            {
+                proc.StartInfo.FileName = _pipePath;
+                proc.StartInfo.Arguments = string.Format(CultureInfo.InvariantCulture, _cmdArgs, commandText);
+                proc.StartInfo.WorkingDirectory = System.IO.Path.GetDirectoryName(_pipePath);
+                proc.EnableRaisingEvents = false;
+                proc.StartInfo.RedirectStandardInput = false;
+                proc.StartInfo.RedirectStandardOutput = true;
+                proc.StartInfo.RedirectStandardError = true;
+                proc.StartInfo.UseShellExecute = false;
+                proc.StartInfo.CreateNoWindow = true;
+                proc.Start();
+                proc.WaitForExit(timeout);
+                exitCode = proc.ExitCode;
+            }
+            finally
```

Comment:
Did some tests, streams are empty when the process did not launch. So there is no reason to read it on finally. 

---
In reply to: [79071150](https://github.com/Microsoft/MIEngine/pull/446#discussion_r79071150) [](ancestors = 79071150)